From faeccc288aa426a219d4b29da43d433de10942e3 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Sun, 20 Feb 2022 16:31:04 +0000
Subject: [PATCH 90/90] WIP: arm64: dts: meson-g12a-x96-max: add support for
 the 7 segment display

First attempt at adding support for the 7 segment display of the device.

See https://github.com/arthur-liberman/vfd-configurations/blob/master/x96-max-1gbit-vfd.conf

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 .../boot/dts/amlogic/meson-g12a-x96-max.dts   | 59 +++++++++++++++++++
 1 file changed, 59 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12a-x96-max.dts b/arch/arm64/boot/dts/amlogic/meson-g12a-x96-max.dts
index b4e86196e346..36ca9c6a1bdf 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12a-x96-max.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12a-x96-max.dts
@@ -8,6 +8,7 @@
 #include "meson-g12a.dtsi"
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/gpio/meson-g12a-gpio.h>
+#include <dt-bindings/leds/common.h>
 #include <dt-bindings/sound/meson-g12a-tohdmitx.h>
 
 / {
@@ -228,6 +229,64 @@
 		};
 	};
 
+	spi {
+		compatible = "spi-gpio";
+		sck-gpios = <&gpio GPIOX_0 GPIO_ACTIVE_HIGH>;
+		mosi-gpios = <&gpio GPIOA_15 GPIO_ACTIVE_HIGH>;
+		cs-gpios = <&gpio_ao GPIOAO_10 GPIO_ACTIVE_LOW>;
+		num-chipselects = <1>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		tm1628: led-controller@0 {
+			compatible = "titanmec,tm1628";
+			reg = <0>;
+			spi-3wire;
+			spi-lsb-first;
+			spi-rx-delay-us = <1>;
+			spi-max-frequency = <500000>;
+			#address-cells = <2>;
+			#size-cells = <0>;
+
+			segment-mapping = /bits/ 8 <4 5 6 1 2 3 7>;
+			grid = /bits/ 8 <3 4 5 1>;
+
+			apps@2,1 {
+				reg = <2 1>;
+				function = "apps";
+			};
+
+			usb@5,2 {
+				reg = <2 2>;
+				function = LED_FUNCTION_USB;
+			};
+			setup@2,3 {
+				reg = <2 3>;
+				function = "setup";
+			};
+
+			card@2,4 {
+				reg = <2 4>;
+				function = LED_FUNCTION_SD;
+			};
+
+			colon@2,5 {
+				reg = <2 5>;
+				function = "colon";
+			};
+
+			hdmi@2,6 {
+				reg = <2 6>;
+				function = "hdmi";
+			};
+
+			cvbs@2,7 {
+				reg = < 7>;
+				function = "cvbs";
+			};
+		};
+	};
+
 	wifi32k: wifi32k {
 		compatible = "pwm-clock";
 		#clock-cells = <0>;
-- 
2.17.1

